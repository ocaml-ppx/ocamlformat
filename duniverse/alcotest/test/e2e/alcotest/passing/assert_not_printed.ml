(** Regression test for an issue in which stderr was not captured into the test
    logs due to Format buffers not being flushed. See
    https://github.com/mirage/alcotest/pull/228 for details. *)

let () =
  let open Alcotest in
  let s tc = tc ^ ": SHOULD NOT BE PRINTED" in
  run __FILE__
    [
      ( "alpha",
        [
          Alcotest.test_case "0" `Quick (fun () ->
              Alcotest.(check unit) (s "0") () ());
          Alcotest.test_case "1" `Quick (fun () ->
              Format.eprintf "%s" (s "1");
              Alcotest.(check unit) (s "1") () ());
        ] );
      ( "beta",
        [
          Alcotest.test_case "2" `Quick (fun () ->
              Alcotest.(check unit) "2" () ();
              Format.eprintf "%s" (s "2"));
        ] );
    ]
