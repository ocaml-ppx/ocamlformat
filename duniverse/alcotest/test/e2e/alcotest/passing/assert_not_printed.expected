Testing `test/e2e/alcotest/passing/assert_not_printed.ml'.
This run has ID `<uuid>'.

  [OK]          alpha          0   0.
  [OK]          alpha          1   1.
  [OK]          beta           0   2.

Full test results in `<build-context>/_build/_tests/<test-dir>'.
Test Successful in <test-duration>s. 3 tests run.
